### DISCUSSION TITLE
PurgeItemsAsync when you only have a non-generic IOfflineTable reference

### DISCUSSION THREAD

**Discussion by oliverw** [Azure/azure-mobile-apps#617] [Q&A]
Is it possible to express the intent to purge soft-deleted entities off a IOfflineTable (non-generic) using the string based query parameter of `IOfflineTable.PurgeItemsAsync` ?

**Comment by adrianhall** [comment]
The string-based query parameter is an OData query; so it should be possible.  However, I have not tested that precise scenario.

**Reply by oliverw** [reply]
> OData query

Any suggestions for a query to start with? I have zero experience with OData.

**Reply by adrianhall** [reply]
That's why we have the typed version and generics

```csharp
await offlineTable.PurgeItemsAsync("$filter=(deleted=true)&__includedeleted=true");
```

should work. You may need to drop the $filter= - there are plenty of OData examples in the test suite.
